auth/auth.go

```diff
@@ -0,0 +1,61 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+)
+
+type AccessToken struct {
+	AccessToken string `json:"access_token"` // Valid for 6hrs
+	// TODO: Use the below value to determine whether or not to make the request
+	ExpiresIn   int32  `json:"expires_in"`
+}
+
+func GetToken(apiUrl, clientId, clientSecret string) (tkn *AccessToken, err error) {
+
+	if len(clientId) == 0 && len(clientSecret) == 0 {
+		log.Fatal(" Both CLIENT_ID and CLIENT_SECRET environment variables are required")
+	}
+
+	audience := fmt.Sprintf("%s/api/v2/", apiUrl)
+
+	url := fmt.Sprintf("%s/oauth/token", apiUrl)
+
+	payload := strings.NewReader(fmt.Sprintf("grant_type=client_credentials&client_id=%s&client_secret=%s&audience=%s", clientId, clientSecret, audience))
+
+	req, err := http.NewRequest("POST", url, payload)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("content-type", "application/x-www-form-urlencoded")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	tkn, err = handleData(body)
+	if err != nil {
+		return nil, err
+	}
+
+	return tkn, nil
+}
+
+func handleData(body []byte) (*AccessToken, error) {
+	var accessToken = AccessToken{}
+	err := json.Unmarshal(body, &accessToken)
+
+	return &accessToken, err
+}
```

Comment:
probably have this in a function is overkill as it's simple enough and you could inline.